src/ro/onlinelibrary/service/ServiceOnlineLibrary.java

```diff
@@ -1,85 +1,418 @@
+        System.out.println(" << List of employees >>");
+        personnel.sort((Personnel p1, Personnel p2) -> p1.getFirstName().compareTo(p2.getFirstName()));
+        personnel.forEach(withCounter((i, person) -> {System.out.println("\n<" + (i+1) + "> " + person);}));
+    }
+
+    @Override
+    public void menu(Scanner scanner, ServiceOnlineLibrary generalAccess, ServiceAudit audit) throws IOException {
+        audit.writeActionToFile("Menu");
+
+        System.out.println("\t << MENU >>");
+
+        while (true) {
+
+            int option1;
+
+            while (true) {
+                System.out.println("\n #OPTIONS:");
+                System.out.println("\t*1* Show books");
+                System.out.println("\t*2* Show upcoming events");
+                System.out.println("\t*3* Show readers");
+                System.out.println("\t*4* Show the employees");
+                System.out.println("\t*5* Upload a book");
+                System.out.println("\t*6* Plan an event");
+                System.out.println("\t*7* Register a reader");
+                System.out.println("\t*8* Upload an employee");
+                System.out.println("\t*9* Exit the menu");
+
+                try {
+                    System.out.println(" #Insert the number of the option:");
+                    option1 = scanner.nextInt();
+                    scanner.nextLine();
+                    break;
+                } catch (Exception exception) {
+                    scanner.nextLine();
+                    System.out.println("\tOptions are between 1 and 9!");
+                }
+            }
+
+            switch (option1) {
+                case (1):
+                    int option2;
+                    while (true) {
+                        System.out.println(" #Books register:");
+                        System.out.println("\t*1* Show all the books");
+                        System.out.println("\t*2* Show books from a certain year");
+                        try {
+                            System.out.println(" #Insert the number of the option:");
+                            option2 = scanner.nextInt();
+                            break;
+                        } catch (Exception exception) {
+                            scanner.nextLine();
+                            System.out.println("\tOptions are between 1 and 2!");
+                        }
+                    }
+                    switch (option2) {
+                        case (1) :
+                            showBooks(audit);
+                            break;
+                        case (2) :
+                            System.out.println(" #Year:");
+                            int publishingYear = scanner.nextInt();
+                            scanner.nextLine();
+                            try {
+                                if (publishingYear <= 0){
+                                    throw new SpecialExceptions("");

```

Comment:
never throw it like this with empty message, put a specific one, it helps you while debugging